planner/core/memtable_predicate_extractor.go

```diff
@@ -535,26 +562,13 @@ func (e *MetricTableExtractor) Extract(
 		return nil
 	}
 
-	// Extract the label columns.
-	for _, name := range names {
-		switch name.ColName.L {
-		case "quantile", "time", "value":
-			continue
-		}
-		var values set.StringSet
-		remained, skipRequest, values = e.extractCol(schema, names, remained, name.ColName.L, false)
-		if skipRequest {
-			e.SkipRequest = skipRequest
-			return nil
-		}
-		if len(values) == 0 {
-			continue
-		}
-		if e.LabelConditions == nil {
-			e.LabelConditions = make(map[string]set.StringSet)
-		}
-		e.LabelConditions[name.ColName.L] = values
+	excludeCols := set.NewStringSet("quantile", "time", "value")
+	remained, skipRequest, extractCols := e.extractCols(schema, names, remained, excludeCols, false)
+	e.SkipRequest = skipRequest
+	if e.SkipRequest {
+		return nil
 	}
+	e.LabelConditions = extractCols
```

Comment:
Should we store `extractCols` to `e.LabelConditions`, even `skipRequest` is true?
For avoiding check `e.LabelConditions == nil` after use this `Extract` function

Reply:
The `cols` always nil if `skipRequest==true`.